db/compaction_job.cc

```diff
@@ -1001,10 +1001,12 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
   RecordDroppedKeys(c_iter_stats, &sub_compact->compaction_job_stats);
   RecordCompactionIOStats();
 
-  if (status.ok() &&
-      (shutting_down_->load(std::memory_order_relaxed) || cfd->IsDropped())) {
-    status = Status::ShutdownInProgress(
-        "Database shutdown or Column family drop during compaction");
+  if (status.ok() && cfd->IsDropped()) {
+    status =
+        Status::ColumnFamilyDropped("Column family dropped during compaction");
+  }
+  if (status.ok() && shutting_down_->load(std::memory_order_relaxed)) {
```

Comment:
That would mean that when **status in not ok** and (status.IsColumnFamilyDropped() == true && shutting_down_->load(std::memory_order_relaxed) == true) then status = Status::ShutdownInProgress("Database shutdown"). Is that acceptable? because in the original code only if status.ok() then the check for shutting_down_->load(std::memory_order_relaxed) or cfd->IsDropped() happens but in what you're proposing status.ok() == false can still eventually lead to status = Status::ShutdownInProgress("Database shutdown"). 
We need to confirm the behavior for 2 possibilities:
1. Status is ok and status.IsColumnFamilyDropped() == true && shutting_down_->load(std::memory_order_relaxed) == true then what is the status?
2. Status is not ok status.IsColumnFamilyDropped() == true && shutting_down_->load(std::memory_order_relaxed) == true then what is the status?
Please let me know and I can change the code appropriately.

Reply:
If Status is OK, then status.IsColumnFamilyDropped must be false.
Original code returns ShutdownInProgress if the db is shutting down **or** the column family is dropped.
In the new code, we **first** check if the column family is dropped. If so, the status becomes `ColumnFamilyDropped`. Next, we check if the db is shutting down. Your original code will skip this check if the column family is dropped.